### DISCUSSION TITLE
Cache when rendering a notebook

### DISCUSSION THREAD

**Discussion by linogaliana** [quarto-dev/quarto-cli#5434] [Q&A]
I need to give access to jupyter notebook to beginners. I am writing that notebook using `.qmd` format and render it into `.ipynb`.

```shell
quarto render index.qmd --to ipynb
```

The notebook can be found [here](https://github.com/InseeFrLab/funathon2023_sujet4/blob/main/index.qmd) and uses some time consuming computations. I would like to skip them when re-rendering the document (to check, for instance, that some elements are okay) and [_caching_](https://quarto.org/docs/computations/caching.html#jupyter-cache) seemed like a good approach. 

I thus put the followiing elements in my header:

```shell
---
title: "Un Yuka avec Python"
execute:
  cache: true
format:
  html:
    code-fold: true
    code-summary: "Dérouler la solution 🟡"
---
```

But each time I re-render the document, everything starts from zero again. Am I using the option wrong ?

**Comment by cderv** [comment]
Does caching works when you rendering to `html` and not `.ipynb` ?  

As caching with python is using jupyter system, and you are rendering to ipynb itself here, it is possible that there is no caching happening. 
I am not clear about the code path for converting to ipynb - this a possible output format, but not sure how much of quarto feature are supported with this output format. 

@jjallaire, do you have a more clear answer on this ? Thank you

**Reply by jjallaire** [reply]
The notebook cache is per-format (as the image formats and sizes can vary across pandoc formats) so you won't hit the cache on your first ipynb render after an HTML render (but you will hit it on subsequent ipynb renders)
